backend/windmill-api/src/jobs.rs

```diff
@@ -1967,3 +1972,56 @@ async fn delete_completed_job(
     tx.commit().await?;
     Ok(Json(job))
 }
+
+#[derive(Debug, sqlx::FromRow, Serialize)]
+struct JobArgs {
+    created_by: String,
+    started_at: DateTime<Utc>,
+    args: serde_json::Value,
+}
+
+async fn get_input_history_by_hash(
+    authed: Authed,
+    Extension(user_db): Extension<UserDB>,
+    Path((w_id, hash)): Path<(String, ScriptHash)>,
+    Query(pagination): Query<Pagination>,
+) -> error::JsonResult<Vec<JobArgs>> {
+    let (per_page, offset) = paginate(pagination);
+
+    let mut tx = user_db.begin(&authed).await?;
+
+    let rows = sqlx::query_as::<_, JobArgs>("select distinct on (args) created_by, started_at, args from completed_job where script_hash = $1 and workspace_id = $2 order by args, started_at desc limit $3 offset $4")
+        .bind(&hash.0)
+        .bind(&w_id)
+        .bind(per_page as i32)
+        .bind(offset as i32)
+        .fetch_all(&mut tx)
+        .await?;
+
+    tx.commit().await?;
+
+    Ok(Json(rows))
+}
+
+async fn get_input_history_by_path(
+    authed: Authed,
+    Extension(user_db): Extension<UserDB>,
+    Path((w_id, hash)): Path<(String, StripPath)>,
+    Query(pagination): Query<Pagination>,
+) -> error::JsonResult<Vec<JobArgs>> {
+    let (per_page, offset) = paginate(pagination);
+
+    let mut tx = user_db.begin(&authed).await?;
+
+    let rows = sqlx::query_as::<_, JobArgs>("select distinct on (args) created_by, started_at, args from completed_job where script_path = $1 and workspace_id = $2 order by args, started_at desc limit $3 offset $4")
```

Comment:
if you were to filter by flow, you will need a new condition in your where. Scripts and flows can have the same path.